hack/.vendor-helpers.sh

```diff
@@ -88,24 +88,18 @@ clean() {
 
 	echo -n 'collecting import graph, '
 	local IFS=$'\n'
-	packages+=( $(
-		for platform in "${dockerPlatforms[@]}"; do
-			export GOOS="${platform%/*}";
-			export GOARCH="${platform##*/}";
-			for buildTags in "${buildTagCombos[@]}"; do
-				go list -e -tags "$buildTags" -f '{{join .Deps "\n"}}' "${packages[@]}"
-			done
-		done | grep -E "^${PROJECT}" | grep -vE "^${PROJECT}/vendor" | sort -u
-	) )
 	local imports=( $(
 		for platform in "${dockerPlatforms[@]}"; do
 			export GOOS="${platform%/*}";
 			export GOARCH="${platform##*/}";
 			for buildTags in "${buildTagCombos[@]}"; do
-				go list -e -tags "$buildTags" -f '{{join .Deps "\n"}}' "${packages[@]}"
-				go list -e -tags "$buildTags" -f '{{join .TestImports "\n"}}' "${packages[@]}"
+				pkgs=( $(go list -e -tags "$buildTags" -f '{{join .Deps "\n"}}' "${packages[@]}" | grep -E "^${PROJECT}" | grep -vE "^${PROJECT}/vendor" | sort -u) )
+				pkgs+=( ${packages[@]} )
+				testImports=( $(go list -e -tags "$buildTags" -f '{{join .TestImports "\n"}}' "${pkgs[@]}" | sort -u) )
+				printf '%s\n' "${testImports[@]}"
```

Comment:
You had me worried!

On Wednesday, September 16, 2015, Tianon Gravi <notifications@github.com>
wrote:

> In hack/.vendor-helpers.sh
> <https://github.com/docker/docker/pull/16294#discussion_r39696176>:
>
> >  	local imports=( $(
> >  		for platform in "${dockerPlatforms[@]}"; do
> >  			export GOOS="${platform%/*}";
> >  			export GOARCH="${platform##*/}";
> >  			for buildTags in "${buildTagCombos[@]}"; do
> > -				go list -e -tags "$buildTags" -f '{{join .Deps "\n"}}' "${packages[@]}"
> > -				go list -e -tags "$buildTags" -f '{{join .TestImports "\n"}}' "${packages[@]}"
> > +				pkgs=( $(go list -e -tags "$buildTags" -f '{{join .Deps "\n"}}' "${packages[@]}" | grep -E "^${PROJECT}" | grep -vE "^${PROJECT}/vendor" | sort -u) )
> > +				pkgs+=( ${packages[@]} )
> > +				testImports=( $(go list -e -tags "$buildTags" -f '{{join .TestImports "\n"}}' "${pkgs[@]}" | sort -u) )
> > +				printf '%s\n' "${testImports[@]}"
>
> Oh interesting, printf appears to just "handle" this for us:
>
> $ printf '%s\n' a b cabc
>
> [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/16294/files#r39696176>.
>